flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/declarative/DeclarativeSchedulerNG.java

```diff
@@ -588,16 +587,12 @@ public void declineCheckpoint(DeclineCheckpoint decline) {
     public void deliverOperatorEventToCoordinator(
             ExecutionAttemptID taskExecution, OperatorID operator, OperatorEvent evt)
             throws FlinkException {
-        state.tryCall(
-                        StateWithExecutionGraph.class,
-                        stateWithExecutionGraph ->
-                                stateWithExecutionGraph.deliverOperatorEventToCoordinator(
-                                        taskExecution, operator, evt),
-                        "deliverOperatorEventToCoordinator")
-                .orElseThrow(
-                        () ->
-                                new TaskNotRunningException(
-                                        "Task is not known or in state running on the JobManager."));
```

Comment:
I think we are changing semantics here. We should still throw an exception if we are not in a state where we can execute it.

Reply:
I will fix this while merging it.